src/components/ceremony/ceremony.edgedb.repository.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import { PaginatedListType, PublicOf, UnsecuredDto } from '~/common';
+import { DbTypeOf } from '~/core';
+import { ChangesOf } from '~/core/database/changes';
+import { castToEnum, RepoFor } from '~/core/edgedb';
+import { CeremonyRepository } from './ceremony.repository';
+import {
+  Ceremony,
+  CeremonyListInput,
+  CeremonyType,
+  CreateCeremony,
+  UpdateCeremony,
+} from './dto';
+
+@Injectable()
+export class CeremonyEdgeDBRepository
+  extends RepoFor(Ceremony, {
+    hydrate: (ceremony) => ({
+      ...ceremony['*'],
+      engagement: true,
+      type: castToEnum(ceremony.__type__.name.slice(12, -8), CeremonyType),
+    }),
+  }).customize((cls, { defaults }) => {
+    return class extends cls {
+      static omit = [defaults.readOne, defaults.readMany];
+      async create(input: CreateCeremony): Promise<any> {
+        // EdgeDB creates these ceremonies in the process of creating the parent
+        return;
+      }
+      async update(
+        existing: DbTypeOf<Ceremony>,
+        changes: ChangesOf<Ceremony, UpdateCeremony>,
+      ): Promise<DbTypeOf<Ceremony>> {
+        return await this.defaults.update({ existing, changes });
```

Comment:
TS errors I am struggling to make sense of here and at the same place in `list` too. @CarsonF

Reply:
As I mentioned above, this should just be
```ts
async update(changes: UpdateCeremony)
```
to match the new neo4j signature you have.
Which does in fact match the default signature here.
So we don't need to declare.

----

On that note, you are not actually changing the signature of `create` either.
So you don't actually need to `customize()` you can just `override` in the final class.
```ts
class CerRepo extends RepoFor().withDefaults() {
  override create() {
    // noop
  }
}
```

Reply:
So what you're saying here is this should look like:
```
@Injectable()
export class CeremonyEdgeDBRepository
  extends RepoFor(Ceremony, {
    hydrate: (ceremony) => ({
      ...ceremony['*'],
      engagement: true,
      type: castToEnum(ceremony.__type__.name.slice(12, -8), CeremonyType),
    }),
  }).withDefaults()
  implements PublicOf<CeremonyRepository> {
    override create() {}
}
```

Reply:
> As I mentioned above, this should just be
> 
> ```ts
> async update(changes: UpdateCeremony)
> ```
> 
> to match the new neo4j signature you have. Which does in fact match the default signature here. So we don't need to declare.
> 
> Same for `list`.
> 
> On that note, you are not actually changing the signature of `create` either. So you don't actually need to `customize()` you can just `override` in the final class.
> 
> ```ts
> class CerRepo extends RepoFor().withDefaults() {
>   override create() {
>     // noop
>   }
> }
> ```

I made this change and still see issues with the return signature. Essentially it seems like one of the `update` is returning `Promise<readonly<entity>>` while the other is returning `readonly<entity>`

Reply:
☝️ Biggest struggle with all of these is knowing which `update` needs to be updated since they are all named the same thing. It's killing me.

Reply:
Ahh... I fixed the DTO as you mentioned above and I think the update is working properly afterwards. That makes sense now too.